apps/render_cf.py

```diff
@@ -158,6 +164,45 @@ def render_templates(job_types, security_environment, api_name):
         template_file.with_suffix('').write_text(output)
 
 
+def get_compute_environments(job_types: dict, compute_env_file: Path) -> list[dict]:
+    compute_envs = []
+    compute_env_names = []
+    compute_env_imports = set()
+    for _, job_spec in job_types.items():
+        for task in job_spec['tasks']:
+            compute_env = task['compute_environment']
+            if 'name' in compute_env:
+                name = compute_env['name']
+                if name in compute_env_names:
+                    raise NameError(
+                        f'Compute envs must have unique names but the following is defined more than once: {name}.'
+                    )
+                compute_envs.append(compute_env)
+                compute_env_names.append(name)
+            elif 'import' in compute_env and compute_env['import'] != 'Default':
+                compute_env_imports.add(compute_env['import'])
+
+    if compute_env_file:
+        compute_envs_from_file = yaml.safe_load(compute_env_file.read_text())['compute_environments']
+        for name in compute_envs_from_file:
+            if name in compute_env_names:
+                raise NameError(
+                    f'Compute envs must have unique names but the following is defined more than once: {name}.'
+                )
+            compute_env = compute_envs_from_file[name]
+            compute_env['name'] = name
+            compute_envs.append(compute_env)
+            compute_env_names.append(name)
+
+    for name in compute_env_imports:
+        if name not in compute_envs_from_file:
+            raise NotImplementedError(
+                f'The following compute env is imported but not defined in the compute envs file: {name}.'
+            )
```

Comment:
I think there's a stronger argument for using `NotImplementedError` here than there is for using `NameError` in the examples above, but the docstring for `NotImplementedError` says `Method or function hasn't been implemented yet`, so I feel like it's a builtin exception that's only intended to be raised when you call a function or method that has been defined but hasn't yet been implemented.

In general, I think builtin exceptions have pretty specific intended use cases and it's usually better to define a custom exception when `ValueError` isn't specific enough (or just use `assert` for non-user-facing code).
```suggestion
            raise ValueError(
                f'The following compute env is imported but not defined in the compute envs file: {name}.'
            )
```